subject_MVC14/subject_14/Controllers/ViewController.swift

```diff
@@ -10,34 +10,45 @@ import UIKit
 final class ViewController: UIViewController {
 
     @IBOutlet private weak var tableView: UITableView!
-    private let fruitsArray = FruitsArray()
+
+    private var fruitsArray: [Fruit] = []
+
+    private static let initialFruitsArray = [
+        Fruit(isChecked: false, name: "りんご"),
+        Fruit(isChecked: true, name: "みかん"),
+        Fruit(isChecked: false, name: "バナナ"),
+        Fruit(isChecked: true, name: "パイナップル")
+    ]
+
     private let fruitsArrayRepository = FruitsArrayRepository()
 
     override func viewDidLoad() {
         super.viewDidLoad()
         tableView.register(CustomTableViewCell.nib(), forCellReuseIdentifier: CustomTableViewCell.identifier)
         tableView.dataSource = self
         tableView.tableFooterView = UIView()
-        self.fruitsArray.fruits = self.fruitsArrayRepository.load()
+        fruitsArray = fruitsArrayRepository.load() ?? Self.initialFruitsArray
     }
 
     @IBAction func addCellDidTapped(_ sender: UIBarButtonItem) {
-        let modalViewController = ModalViewController.instantiate(
+        let inputViewController = InputViewController.instantiate(
             didSaveFruits: { [weak self] text in
+                guard let strongSelf = self else { return }
                 // 【疑問】なぜselfを明示しないといけないのか, なぜself!でもいいのか、nilの場合エラーの可能性があるではないか
```

Comment:
> Normally, a closure captures variables implicitly by using them in the body of the closure, but in this case you need to be explicit. If you want to capture self, write self explicitly when you use it, or include self in the closure’s capture list. Writing self explicitly lets you express your intent, and reminds you to confirm that there isn’t a reference cycle.

Closures — The Swift Programming Language (Swift 5.5)
https://docs.swift.org/swift-book/LanguageGuide/Closures.html